pkg/controller/worker/machines.go

```diff
@@ -151,6 +154,12 @@ func (w *workerDelegate) generateMachineConfig(_ context.Context) error {
 				disk["interface"] = *workerConfig.Volume.LocalSSDInterface
 			}
 
+			if workerConfig.Volume != nil && *volume.Type != "SCRATCH" {
```

Comment:
I think you should also check for `volume.Type != nil` here

Reply:
yep thats right. fixed.